### PR TITLE
Pro 3501 expand placeholder

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR adds documentation to clarify how to add image and video placeholder content to custom widgets. Closes PRO-3501.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

1. Build the documentation.
2. Check for the new page under 'Advanced topics -> Other customizations'

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: docs/guide/adding-custom-widget-placeholder-content.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Adding placeholders to custom widgets
+
+Custom widgets can also take advantage of non-schema field placeholders by adding code to the main module, preparing the widget template to show the custom placeholder content, and adding the placeholder content into an accessible folder.
```
Comment (boutell) [inline docs/guide/adding-custom-widget-placeholder-content.md:3]: "main module" is confusing. How about just "the widget module in question"

File: docs/guide/adding-custom-widget-placeholder-content.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+## Basic placeholder addition
+
+### Altering the `index.js` file
+
+ In order to add a custom image file, two options should be set within the main widget file, `placeholder: true` and the `placeholderImage` option with the value set to the file extension of the placeholder content without any `.` prefix.
```
Comment (boutell) [inline docs/guide/adding-custom-widget-placeholder-content.md:9]: the module's `index.js` file

Comment (boutell) [reply docs/guide/adding-custom-widget-placeholder-content.md:9]: (main widget file is in the eye of the beholder IMHO)

File: docs/guide/adding-custom-widget-placeholder-content.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+</AposCodeBlock>
+
+::: note
+You can bypass the call to `determineBestAsetUrl()` for your images or videos stored in the `public` folder and instead pass the path to the asset using `placeholderUrl` in place of `placeholderImage`. This is discouraged because errors can be made in this path. It is better to let Apostrophe figure this out for you.
+:::
+
+### Altering the widget.html file
+
+The specific alteration of the `widget.html` template will depend on the type of placeholder content being delivered. Irrespective of file type, the template will have access to `data.widget.aposPlaceholder` and `data.manager.options.placeholderUrl` to populate the markup.
```
Comment (boutell) [inline docs/guide/adding-custom-widget-placeholder-content.md:46]: "But I never set `options.placeholderUrl`!" It is populated for you when you set `placeholderImage` and call `determineBestAssetUrl`.

File: docs/guide/adding-custom-widget-placeholder-content.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+```twig
+<section data-custom-widget>
+  <h1>Custom Widget</h1>
+  {% if data.widget.aposPlaceholder and data.manager.options.placeholderUrl %}
+  <img
+    src="{{ data.manager.options.placeholderUrl }}"
+    alt="{{ __t('nameSpace:imagePlaceholder') }}"
+    class="custom-widget-placeholder"
+  />
+  {% endif %}
```
Comment (boutell) [inline docs/guide/adding-custom-widget-placeholder-content.md:65]: "else: render the widget normally"

File: docs/guide/adding-custom-widget-placeholder-content.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+</section>
+```
+<template v-slot:caption>
+  modules/custom-widget/views/widget.html
+</template>
+
+</AposCodeBlock>
+
+This example demonstrates adding a video. Depending on where the video is delivered from, you can either take advantage of the `@apostrophecms/oembed` and `@apostrophecms/video-widget` module's abilities to handle these types of content automatically. For project-uploaded videos, you will need to create appropriate markup.
```
Comment (boutell) [inline docs/guide/adding-custom-widget-placeholder-content.md:74]: Your example here is for a self-hosted video that's a static asset. Let's just stick to that. I think the discussion of using the oembed module or video-widget is confusing.

**Review by boutell — CHANGES_REQUESTED** [review]

**Comment by boutell** [comment]
A larger point here is that "you can do it yourself": you can look at `widget.aposPlaceholder` and decide to do whatever you want. But I think these are great examples that should make that clear to those who would think to try it.

File: docs/guide/adding-custom-widget-placeholder-content.md (hunk 1 above)
Comment (BoDonkey) [reply docs/guide/adding-custom-widget-placeholder-content.md:3]: Changed.

File: docs/guide/adding-custom-widget-placeholder-content.md (hunk 2 above)
Comment (BoDonkey) [reply docs/guide/adding-custom-widget-placeholder-content.md:9]: Yeah, that is what I was struggling with. People like Greg seem to have an index.js file that just imports from a bunch of other files. Didn't want to alienate people who program like that.

File: docs/guide/adding-custom-widget-placeholder-content.md (hunk 3 above)
Comment (BoDonkey) [reply docs/guide/adding-custom-widget-placeholder-content.md:46]: Changed the third paragraph of the section to better clarify.

File: docs/guide/adding-custom-widget-placeholder-content.md (hunk 4 above)
Comment (BoDonkey) [reply docs/guide/adding-custom-widget-placeholder-content.md:65]: Changed.

File: docs/guide/adding-custom-widget-placeholder-content.md (hunk 5 above)
Comment (BoDonkey) [reply docs/guide/adding-custom-widget-placeholder-content.md:74]: Changed.

**Review by boutell — APPROVED** [review]

**Pull request merged**